engine/src/main/java/org/terasology/engine/world/block/BlockArea.java

```diff
@@ -354,4 +355,35 @@ public int hashCode() {
     public String toString() {
         return "BlockArea[(" + this.minX + ", " + this.minY + ")...(" + this.maxX + ", " + this.maxY + ")]";
     }
+
+    private int calDis(int x, int y){
+        return (int) Math.sqrt(x * x + y * y);
+    }
+    public int distanceSquared(int px, int py){
+        if(px < minX){
+            if(py < minY) {
+                return calDis(minX - px, minY - py);
+            }
+            if(py <= maxY) {
+                return minX - px;
+            }
+            return calDis(minX - px, maxY - py);
+        }else if(px <= maxX){
+            if(py < minY) {
+                return minY - py;
+            }
+            if(py <= maxY) {
+                return 0;
+            }
+            return py - maxY;
+        }else{
+            if(py < minY) {
+                return calDis(maxX - px, minY - py);
+            }
+            if(py <= maxY) {
+                return px - maxX;
+            }
+            return calDis(maxX - px, maxY - py);
+        }
```

Comment:
this logic doesn't look quite right to me. 

here is what is defined in the current Rectanglei class. the current logic in Rectanglei is pretty concise and it works well. 

```
        //     center = min + width / 2 = min + (max - min) / 2
        // 2 * center = 2 * min + max - min = min + max
        int cx2 = minX() + maxX();
        int cy2 = minY() + maxY();

        //     dx = p - center - width / 2
        // 2 * dx = 2 * px - 2 * center - width
        //     dx = (2 * px - 2 * center - width) / 2
        long dx = Math.max(Math.abs(2 * px - cx2) - getSizeX(), 0) / 2;
        long dy = Math.max(Math.abs(2 * py - cy2) - getSizeY(), 0) / 2;
        return dx * dx + dy * dy;

```

Reply:
Yes, I shall be making the required changes.